Mathlib/MeasureTheory/Measure/Lebesgue/VolumeOfBalls.lean

```diff
@@ -0,0 +1,448 @@
+  induction n with
+  | zero =>
+      simp only [Nat.zero_eq, volume_pi, Finset.univ_eq_empty, Finset.prod_empty, integral_const,
+        pi_empty_univ, ENNReal.one_toReal, smul_eq_mul, mul_one, pow_zero]
+  | succ n n_ih =>
+      calc
+        _ = ∫ x : E × (Fin n → E), (f x.1) * ∏ i : Fin n, f (x.2 i) := by
+          rw [volume_pi, ← ((measurePreserving_piFinSuccAboveEquiv
+            (fun _ => (volume : Measure E)) 0).symm).integral_comp']
+          simp_rw [MeasurableEquiv.piFinSuccAboveEquiv_symm_apply, Fin.insertNth_zero',
+            Fin.prod_univ_succ, Fin.cons_zero, Fin.cons_succ]
+          rfl
+        _ = (∫ x, f x) * (∫ x, f x) ^ n := by rw [← n_ih, ← integral_prod_mul, volume_eq_prod]
+        _ = (∫ x, f x) ^ n.succ := by rw [← pow_succ]
+
+theorem MeasureTheory.integral_finset_prod_eq_pow {E : Type*} (ι : Type*) [Fintype ι] (f : E → ℝ)
+    [MeasureSpace E] [SigmaFinite (volume : Measure E)] :
+    ∫ x : ι → E, ∏ i, f (x i) = (∫ x, f x) ^ (card ι) := by
+  let p := measurePreserving_piCongrLeft (fun _ => (volume : Measure E)) (equivFin ι)
+  rw [volume_pi, ← (p.symm).integral_comp', ← integral_finset_prod_eq_pow']
+  congr!
+  rw [Fintype.prod_equiv (equivFin ι)]
+  exact fun _ => rfl
+
+end move_me
+
+namespace MeasureTheory
+
+open MeasureTheory.Measure FiniteDimensional ENNReal Filter
+
+open scoped Topology
+
+noncomputable section main_result
+
+theorem measure_unitBall_eq_integral_div_gamma {E : Type*} {p : ℝ}
+    [NormedAddCommGroup E] [NormedSpace ℝ E] [FiniteDimensional ℝ E] [MeasurableSpace E]
+    [BorelSpace E] [Nontrivial E] (μ : Measure E) [IsAddHaarMeasure μ] (hp : 0 < p) :
+    μ (Metric.ball 0 1) =
+      .ofReal ((∫ (x : E), Real.exp (- ‖x‖ ^ p) ∂μ) / Real.Gamma (finrank ℝ E / p + 1)) := by
+  have : (0:ℝ) < finrank ℝ E := Nat.cast_pos.mpr finrank_pos
+  have : ((∫ y in Set.Ioi (0:ℝ), y ^ (finrank ℝ E - 1) • Real.exp (-y ^ p)) /
+      Real.Gamma ((finrank ℝ E) / p + 1)) * (finrank ℝ E) = 1 := by
+    simp_rw [← Real.rpow_nat_cast _ (finrank ℝ E - 1), smul_eq_mul, Nat.cast_sub finrank_pos,
+      Nat.cast_one]
+    rw [integral_rpow_mul_exp_neg_rpow hp (by linarith), sub_add_cancel,
+      Real.Gamma_add_one (ne_of_gt (by positivity))]
+    field_simp; ring
+  rw [integral_fun_norm_addHaar μ (fun x => Real.exp (- x ^ p)), nsmul_eq_mul, smul_eq_mul,
+    mul_div_assoc, mul_div_assoc, mul_comm, mul_assoc, this, mul_one, ofReal_toReal]
+  exact ne_of_lt measure_ball_lt_top
+
+variable {E : Type*} [AddCommGroup E] [Module ℝ E] [FiniteDimensional ℝ E] [mE : MeasurableSpace E]
+  [tE : TopologicalSpace E] [TopologicalAddGroup E] [BorelSpace E] [T2Space E] [Nontrivial E]
+  [ContinuousSMul ℝ E] (μ : Measure E) [IsAddHaarMeasure μ] {g : E → ℝ} (h1 : g 0 = 0)
+  (h2 : ∀ x, g (- x) = g x) (h3 : ∀ x y, g (x + y) ≤ g x + g y) (h4 : ∀ {x}, g x = 0 → x = 0)
+  (h5 :  ∀ r x, g (r • x) ≤ |r| * (g x))
+
+theorem measure_lt_one_eq_integral_div_gamma {p : ℝ} (hp : 0 < p) :
```

Comment:
You assume `[Nontrivial E]`, but I have the impression that the result is also true for trivial `E`. Same thing in the previous theorem, by the way.

Reply:
Indeed. Thanks for pointing this out. I do need results from another PR #8423 though